3.js

```diff
@@ -0,0 +1,11 @@
+module.exports = function (string, index) {
+  const stringArray = string.split('')
+
+  noPunctuationArray = stringArray.filter(character =>
+    /\w/.test(character) || character === ' '
+  )
+
+  const noPunctuationString = noPunctuationArray.join('').split(' ')
```

Comment:
It joins an array of characters (including spaces) into a string and then splits the string (by spaces) into words.  My name for `noPunctuationString` was totally wrong since it's actually an array.

I'm refactoring this into functions to make it more clear.